packages/blaze-query/src/hydra.ts

```diff
@@ -0,0 +1,758 @@
+        );
+      }
+
+      ret.push(utxo);
+    }
+
+    return ret;
+  }
+
+  /**
+   * This method takes a datum hash and tries to find the assosciated inline datum in the current snapshot utxo set.
+   * If the output containing the datum has been spent, this method will error
+   * @param datumHash
+   * @returns Promise<PlutusData>
+   */
+  async resolveDatum(datumHash: DatumHash): Promise<PlutusData> {
+    const utxos = await this.fetchUTxOs();
+
+    for (const key in utxos) {
+      const utxo = utxos[key]!;
+      const datum = utxo.output().datum();
+      if (!datum) continue;
+      if (datum.kind() != DatumKind.InlineData) continue;
+
+      const hash = datum.asInlineData()?.hash();
+      if (datumHash === hash) {
+        return datum.asInlineData()!;
+      }
+    }
+
+    throw new Error(
+      "resolveDatum: could not find a datum with that hash in the current snapshot",
+    );
+  }
+
+  async awaitTransactionConfirmation(
+    txId: TransactionId,
+    timeout?: number,
+  ): Promise<boolean> {
+    const startTime = Date.now();
+
+    const delay = (ms: number) =>
+      new Promise((resolve) => setTimeout(resolve, ms));
+
+    const checkConfirmation = async () => {
+      const resp = await fetch(`${this.url}snapshot/utxo`);
+      const rawUtxo = (await resp.json()) as any;
+      return !!Object.keys(rawUtxo).find(
+        (txRef) => txRef.split("#")[0] === txId,
+      );
+    };
+
+    if (await checkConfirmation()) {
+      return true;
+    }
+
+    if (timeout) {
+      while (Date.now() - startTime < timeout) {
+        await delay(500);
+
+        if (await checkConfirmation()) {
+          return true;
+        }
+      }
+    }
+
+    return false;
+  }
+
+  async postTransactionToChain(tx: BlazeTransaction): Promise<TransactionId> {
+    const txId = tx.body().hash();
+    this.connection.send(
+      JSON.stringify({
+        tag: "NewTx",
+        transaction: {
+          type: "Tx BabbageEra",
+          cborHex: tx.toCbor(),
+        },
+      }),
+    );
+
+    return txId;
+  }
+  evaluateTransaction(
+    _tx: BlazeTransaction,
+    _additionalUtxos: TransactionUnspentOutput[],
+  ): Promise<Redeemers> {
+    throw new Error("Unsupported by the Hydra API");
+  }
+
+  async fetchUTxOs(): Promise<{ [txRef: string]: TransactionUnspentOutput }> {
+    const utxos: { [txRef: string]: TransactionUnspentOutput } = {};
+    const resp = await fetch(`${this.url}snapshot/utxo`);
+    const rawUtxo = (await resp.json()) as any;
+    for (const key in rawUtxo) {
+      const [txHash, idx] = key.split("#");
+      const output = rawUtxo[key];
+      utxos[key] = this.hydraUtxoToTransactionUnspentOutput(
+        txHash!,
+        parseInt(idx!),
+        output,
+      );
+    }
+
+    return utxos;
+  }
+  private hydraUtxoToTransactionUnspentOutput(
+    txHash: string,
+    idx: number,
+    output: any,
+  ): TransactionUnspentOutput {
```

Comment:
Has the `TransactionUnspentOutput` used here a blaze-specific JSON encoding or can we find a common denominator between Hydra and Blaze? We long wanted to adopt a widely accepted standard (We like https://github.com/CardanoSolutions/cardanonical/blob/main/cardano.json), but have been following the format used by `cardano-cli` - which likely is not the majority format!?

Reply:
It's inherited from https://github.com/input-output-hk/cardano-js-sdk